packages/api/src/external/commonwell/patient-external-data.ts

```diff
@@ -1,12 +1,44 @@
+import { MetriportError } from "@metriport/core/util/error/metriport-error";
+import { executeWithRetries } from "@metriport/shared";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
 import { cloneDeep } from "lodash";
 import { getPatientOrFail } from "../../command/medical/patient/get-patient";
 import { Patient } from "../../domain/medical/patient";
 import { PatientModel } from "../../models/medical/patient";
 import { executeOnDBTx } from "../../models/transaction-wrapper";
 import { LinkStatus } from "../patient-link";
-import { getCQLinkStatus } from "./patient";
+import { getLinkStatusCQ, getLinkStatusCW } from "./patient";
 import { CQLinkStatus, PatientDataCommonwell } from "./patient-shared";
 
+dayjs.extend(duration);
+
+const maxAttemptsToGetPatientCWData = 3;
+const waitTimeBetweenAttemptsToGetPatientCWData = dayjs.duration(2, "seconds");
+
+export type PatientWithCWData = Patient & {
+  data: { externalData: { COMMONWELL: PatientDataCommonwell } };
+};
+
+export async function getPatientWithCWData(patient: Patient): Promise<PatientWithCWData> {
+  const getPatientWithCWDataOrFail = async () => {
+    const patientDB: Patient = await getPatientOrFail({
+      id: patient.id,
+      cxId: patient.cxId,
+    });
+    if (getLinkStatusCW(patientDB.data.externalData) !== "completed") {
+      throw new MetriportError(`Patient is not linked to CW`);
+    }
+    return patientDB as PatientWithCWData;
```

Comment:
should we make this so that if the status `=== "failed"` then we break out of the retry?

ie if it failed in the first second, we could save 5s of waiting, as we know it will stay in that state

Reply:
https://github.com/metriport/metriport-internal/issues/1291

Reply:
I would also say that my PR that is addressing the relinking logic should address this too